BeeminderWidget/BeeminderWidgetConfigurationIntents.swift

```diff
@@ -0,0 +1,51 @@
+//
+//  BeeminderWidgetConfigurationIntents.swift
+//  BeeminderWidget
+//
+//  Created by krugerk on 2024-11-13.
+//
+
+import AppIntents
+import BeeKit
+import WidgetKit
+
+struct GoalBasicsConfigurationAppIntent: WidgetConfigurationIntent {
+    static let title: LocalizedStringResource = "Goal Basics"
+    static let description = IntentDescription("Shows basic data of a goal!")
+
+    @Parameter(title: "Goal Name (aka slug)",
+               default: "steps",
+               inputOptions: String.IntentInputOptions(keyboardType: .default, capitalizationType: .none, autocorrect: false))
+    var goalName: String
```

Comment:
currently requires user to manually type in the goal name. 

- [ ] provide a dropdown list of the goals

Reply:
Yeah, I've been thinking it would make sense to add a Goal AppEntity to be used for widget configuration and also for App Intents (and replacing the existing "Add Data" App Intent with one that uses it). Perhaps that would make sense as a separate PR to merge before this one?

Reply:
initial implementation in https://github.com/beeminder/BeeSwift/pull/515/commits/94262a34c28d197d483ad4b4dcf30b49e495be2e